pkg/vectorize/cot/cot.go

```diff
@@ -118,7 +118,13 @@ func cotFloat64(xs []float64, rs []float64) []float64 {
 
 func Cot[T constraints.Integer | constraints.Float](inputValues []T, resultValues []float64) []float64 {
 	for i, n := range inputValues {
-		resultValues[i] = 1.0 - math.Tan(float64(n))
+		if math.Tan(float64(n)) < 1e-9 && math.Tan(float64(n)) >= 0 {
```

Comment:
redundant computation of `Tan`

Reply:
We need to think about the corner condition, So I add the if .. else..

Reply:
Ok ,I fix it